meshtal_analysis.py

```diff
@@ -325,106 +406,56 @@ def count_zeros(mesh):
     return count
 
 
-def find_mesh_tally_numbers(data):
-    """ find the different meshes in the file, the tally number
-        and the line it starts on"""
-    tdict = {}
-    for i, l in enumerate(data):
-        if "Mesh Tally Number" in l:
-            talid = int(l.split(" ")[-1])
-            tdict[talid] = i
-    return tdict
-
-
-def find_next_mesh(tnum, tdict):
-    """ finds the start location of the next numerical mesh tally"""
-    keylist = sorted(tdict.keys())
-    if tnum == keylist[-1]:
-        return -1
-    else:
-        for i, v in enumerate(keylist):
-            if v == tnum:
-                return tdict[keylist[i+1]]
-
-
-def read_mesh(tnum, data, tdict):
-    """ reads and individual mesh tally into mesh class """
+def read_mesh(path):

```

Comment:
need to add doc string, perhaps eneds to change function name to read read_mesh_tally_file for backwards compatibilty and to reflect what the function does